pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -277,11 +272,16 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.WaitUntilWorkerReady),
 			Dependencies: flow.NewTaskIDs(deployWorker),
 		})
-		_ = g.Add(flow.Task{
-			Name:         "Ensuring ingress DNS record",
-			Fn:           flow.TaskFn(botanist.EnsureIngressDNSRecord).DoIf(dnsEnabled && managedExternalDNS && !o.Shoot.HibernationEnabled).RetryUntilTimeout(defaultInterval, 10*time.Minute),
+		nginxLBReady = g.Add(flow.Task{
+			Name:         "Waiting until nginx ingress LoadBalancer is ready",
+			Fn:           flow.TaskFn(botanist.WaitUntilNginxIngressServiceIsReady).DoIf(botanist.Shoot.NginxIngressEnabled()),
 			Dependencies: flow.NewTaskIDs(deployManagedResources),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Deploying nginx ingress domain record",
+			Fn:           flow.TaskFn(botanist.DeployNginxIngressDNSEntry),
```

Comment:
I had the same comment about the naming. Ref https://github.com/gardener/gardener/pull/1980#discussion_r400883023.

Reply:
Oh sorry @ialidzhikov I missed that :(

Reply:
+1, agreed as well. I'm also missing the `dnsEnabled` check.

Reply:
> Why doesn't it have a DoIf(botanist.Shoot.NginxIngressEnabled()) like the depending step?

It's not needed - by default, when `botanist.DeployNginxIngressDNSEntry` is called it would actually try to remove the dns entry.

Only if DNS is enabled and after the LB in the Shoot cluster gets its IP then the correct value is set https://github.com/gardener/gardener/blob/1e120c562fbfe786a2ef24d509dae4f07d27e40d/pkg/operation/botanist/addons.go#L81-L97